duckduckgo/Android
# Feature/add home screen

Asana Issue URL: https://app.asana.com/0/471012189846020/364465928695404

## Description
First pass at a home screen / browser landing page.

There are a few unanswered questions around navigation and UX that will likely result in further changes. However, as discussed, let's get this implementation in now while we're working out the answers to those questions separately. 

We'll _likely_ want to move the two separate Activities (BrowserActivity and HomeActivity) into a single Activity in future, and treat the _landing page_ and _browser view_ as two separate states within.


## Steps to Test this PR:
1. Run the app; note the new home screen
1. Tap the EditText; ensure you transition to allow you to enter your query
1. Test back navigation works as expected (for now, meaning you **can** return to the landing page)
1. Test that if you enter a search query (not a full URL) then that search query is shown and remains editable in the omnibar.
1. Test that once you navigate to a search result (or enter a full URL) the omnibar updates to show the URL and not a search term
